### ISSUE TITLE
Consider upgrading coffee-jshint's dependencies.

### ISSUE THREAD

**Issue by cueedee** [Clever/coffee-jshint#20]
Hi, 

Please consider upgrading all `coffee-jshint`'s dependencies. Most notably, [`jshint`'s latest](https://github.com/jshint/jshint/releases) (now at `2.8.0`) doesn't match the `~2.5.1` range that `coffee-jshint` currently requires.

Also, the `peerDependencies` list a `>= 1.6 <= 1.8.0` version range for `coffee-script` which excludes [its latest incarnation](https://github.com/jashkenas/coffeescript/releases) (currently at `2.9.3`); this particularly, makes `npm` very unhappy:

```
npm ERR! node v0.12.5
npm ERR! npm  v2.12.0
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package coffee-script does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer coffee-jshint@0.2.2 wants coffee-script@>= 1.6 <= 1.8.0
```

The latter point is also the subject of #17.

**Comment by jonahkagan** [comment]
@cueedee would you be willing to submit a pull request with the upgrade of jshint? Let's address the coffee-script upgrade separately in #17

**Comment by cueedee** [comment]
Done.

**Comment by cueedee** [comment]
Meanwhile, I encountered a number of additional issues that I would like to mention fyi:

* `npm` issues a warning about the project's license:

  ```
  npm WARN package.json coffee-jshint@0.2.2 license should be a valid SPDX license expression
  ```

  The `package.json` specifies `'BSD'` for its `"license"`, which indeed is not a valid [SPDX license expression](http://spdx.org/licenses/). I thought of fixing the exact BSD type too, but found that your `LICENSE` file actually specifies the `'Apache-2.0'` license instead, representing a conflict I feel unqualified to resolve.

* It appears that `optimist`, one of your other `dependencies`, has been discontinued: see: https://www.npmjs.com/package/optimist#deprecation-notice.

* As also mentioned in pr #21, `coffee-script`'s `package.json` these days specifies `"preferGlobal": true` which an `npm install` of `coffee-jshint` warns about when encountering a recent `coffee-script` in the 'devDependencies`:
  ```
  npm ERR! peerinvalid The package coffee-script does not satisfy its siblings' peerDependencies requirements!
  ```

**Comment by jonahkagan** [comment]
Thanks for pointing out the license issue. Fixed in 089c3e9
